### ISSUE TITLE
[Bug] Unified deterministic extend attention uses different xAI temperature scaling than regular attention

### ISSUE THREAD

**Issue by rohanwinsor** [sgl-project/sglang#32942]
### Checklist

- [x] I searched related issues but found no solution.
- [x] The bug persists in the latest version.
- [x] Issues without environment info and a minimal reproducible demo are hard to resolve and may receive no feedback.
- [x] If this is not a bug report but a general question, please start a discussion at https://github.com/sgl-project/sglang/discussions. Otherwise, it will be closed.
- [x] Please use English. Otherwise, it will be closed.

### Describe the bug

The unified deterministic extend attention kernel appears to use different
`xai_temperature_len` scaling from the regular Triton extend/decode attention
paths.

Grok wires `config.attn_temperature_len` into `xai_temperature_len`. The regular
extend attention kernel and both decode attention kernels apply logarithmic
scaling after the configured threshold:

```text
1.0 when position <= xai_temperature_len
log2(position) / log2(xai_temperature_len) when position > xai_temperature_len
```

The unified deterministic extend attention kernel currently applies a
reciprocal scale instead:

```text
1.0 when position < xai_temperature_len
xai_temperature_len / (position + 1) when position >= xai_temperature_len
```

This means enabling the unified deterministic extend path can change Grok/xAI
attention logits compared with the regular Triton path for the same model
configuration.

This looks unintended because regular extend attention and both decode attention
paths agree on the logarithmic formula, while the unified-vs-regular test did
not cover `xai_temperature_len > 0`.

### Reproduction

The existing registered test
`test/registered/attention/test_triton_attention_kernels.py::TestTritonAttention::test_extend_attention_unified_vs_regular`
already checks that unified extend attention matches regular two-stage extend
attention. Extend that test so it also passes `xai_temperature_len=4` into both
kernel calls.

On current main before the fix, a focused CUDA/Triton repro gives:

```text
NVIDIA GeForce RTX 5060 Ti
xai_temperature_len=-1 max_diff=0.000244 allclose=True
xai_temperature_len=4 max_diff=0.827637 allclose=False
```

Expected behavior:

```text
xai_temperature_len=-1 allclose=True
xai_temperature_len=4 allclose=True
```

PR #32943 adds this regression coverage and aligns the unified kernel formula
with regular extend/decode attention.

### Environment

Environment used for the focused CUDA/Triton repro:

```text
Python: 3.12.13
CUDA available: True
GPU 0: NVIDIA GeForce RTX 5060 Ti
GPU 0 Compute Capability: 12.0
CUDA version: 13.0
PyTorch: 2.13.0+cu130
triton: 3.7.1
sglang-kernel: 0.4.5
```
